### ISSUE TITLE
VULNERABILITY: Unused SetID command can be abused to account jam and allow moderator access

### ISSUE THREAD

**Issue by MikeDev101** [meower-media-co/Meower-Server#21]
Re: DM from @tnix100 

TL;DR, This is a security vulnerability found in the Set ID command, originally used for unmodified CloudLink server users to specify a username. Since SetID is not used in Meower as it has been replaced with AutoID, it's functionality still remains, leading to this vuln...

---

**How are you able to pull off this vulnerability?**

By basicly logging into Meower with a valid account and waiting to gain the server's trust, then at the right moment setting your ID to be the ID of another account.

**What will this security vulnerability let you achieve?**

Depending on the timing it will have different outcomes.
1. It adds a "ghost client" to the ulist which cannot be logged out and causes the account to be "account jammed" (owner of account cannot login and will stay in ulist until server restarts).
2. It will replace your proper client's ID with the ID you chose to spoof, allowing you to gain full access over their account and even use moderator level commands if it is a moderator level account.

**What actions did we (as in 'we' I mean William and I) take to protect Meower and it's users?**

1. We deleted the DNS records for server.meower.org so no new traffic can enter the Cloudflare Argo tunnel.
2. We tried to tell everyone to logout of Meower, this will cause them to not be able to log back in.
3. We made a maintenance page that replaced app.meower.org to let users know what's going on and direct them to the Discord for updates.
4. We made an announcement about this in updates to be transparent and let the users of Meower know what's going on.

**Potential Fix**

Not sure if this will patch everything related to this bug, but removing the 'setid' command from the Meower server is a good first step.

---

**Solution:** Disabling functionality within CloudLink Server, deeming the following commands...

1. setid
2. gmsg
3. gvar

...Disabled.

**Comment by MikeDev101** [comment]
A patch for this vulnerability will be issued soon. For the time being, server.meower.org has been brought offline.

**Comment by MikeDev101** [comment]
Vuln patched. Servers and clients should be back online.
